quarkus.ls/com.redhat.quarkus.ls/src/test/java/com/redhat/quarkus/services/ApplicationPropertiesCodeActionsTest.java

```diff
@@ -37,9 +37,56 @@ public void codeActionsForUnknownProperties() throws BadLocationException {
 				"";
 		Diagnostic d = d(1, 0, 23, "Unknown property 'quarkus.application.nme'", DiagnosticSeverity.Warning,
 				ValidationType.unknown);
+
 		testDiagnosticsFor(value, d);
 		testCodeActionsFor(value, d,
 				ca("Did you mean 'quarkus.application.name' ?", te(1, 0, 1, 23, "quarkus.application.name"), d));
 	};
 
+	@Test
+	public void codeActionsForUnknownLogLevelValue() throws BadLocationException {
+		String value = "quarkus.log.level=WARNIN";
+		Diagnostic d = d(0, 18, 24, "Invalid enum value: 'WARNIN' is invalid for type java.util.Optional<java.util.logging.Level>", DiagnosticSeverity.Error,
+				ValidationType.value);
+
+		testDiagnosticsFor(value, d);
+		testCodeActionsFor(value, d,
+				ca("Did you mean 'WARNING'?", te(0, 18, 0, 24, "WARNING"), d));
+	};
+
+	@Test
+	public void codeActionsForUnknownEnum() throws BadLocationException {
+		String value = "quarkus.log.syslog.async.overflow=BLACK";
+		Diagnostic d = d(0, 34, 39, "Invalid enum value: 'BLACK' is invalid for type org.jboss.logmanager.handlers.AsyncHandler$OverflowAction", DiagnosticSeverity.Error,
+				ValidationType.value);
+
+		testDiagnosticsFor(value, d);
+		testCodeActionsFor(value, d,
+				ca("Did you mean 'BLOCK'?", te(0, 34, 0, 39, "BLOCK"), d));
+	};
+
+	@Test
+	public void codeActionsForUnknownBoolean() throws BadLocationException {
+		String value = "quarkus.http.cors=fals";
+		Diagnostic d = d(0, 18, 22, "Type mismatch: boolean expected", DiagnosticSeverity.Error,
+				ValidationType.value);
+
+		testDiagnosticsFor(value, d);
+		testCodeActionsFor(value, d,
+				ca("Did you mean 'false'?", te(0, 18, 0, 22, "false"), d));
+	};
+
+	@Test
+	public void codeActionsForReplaceUnknown() throws BadLocationException {
+		String value = "quarkus.log.syslog.async.overflow=unknown-value";
+		Diagnostic d = d(0, 34, 47,
+				"Invalid enum value: 'unknown-value' is invalid for type org.jboss.logmanager.handlers.AsyncHandler$OverflowAction",
+				DiagnosticSeverity.Error,
+				ValidationType.value);
+
+		testDiagnosticsFor(value, d);
+		testCodeActionsFor(value, d,
+				ca("Replace with 'BLOCK'?", te(0, 34, 0, 47, "BLOCK"), d),
+				ca("Replace with 'DISCARD'?", te(0, 34, 0, 47, "DISCARD"), d));
+	};
```

Comment:
Add a test with the mapped property `quarkus.log.category."org.acme".level=WARNIN`